services/proctoring-service/ai-service/main.py

```
@@ -57,42 +88,92 @@ def extract_features(frame, last_gray, head_angles, yolo_results):
+            ml_pred = clf.predict(X_pred)[0]
+            if ml_pred != 'no_violation':
+                event_map = {"Roi_manhinh": "FACE_NOT_DETECTED", "Nhieu_nguoi": "MULTIPLE_FACES", "Dien_thoai": "MOBILE_PHONE_DETECTED", "Che_camera": "CAMERA_TAMPERED", "Khac": "SUSPICIOUS_BEHAVIOR"}
+                event_type = event_map.get(ml_pred, "SUSPICIOUS_BEHAVIOR")
+                if not any(e['event_type'] == event_type for e in detected_events):
+                    detected_events.append({"event_type": event_type, "severity": "medium", "metadata": {"source": "ml_model", "prediction": ml_pred}})
+
+        print(f"[AI SERVICE] ✅ Phân tích hoàn tất, phát hiện {len(detected_events)} sự kiện")
+        return {"events": detected_events}
+    except HTTPException:
+        raise
+    except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)